### PR TITLE
Enhance Nodes Report to include Policy information

### PR DESCRIPTION
**Pull request by mudash**
Signed-off-by: Mudassar Shafique <15931574+mudash@users.noreply.github.com>

## Description

Adds policy information in CSV and Txt outputs for nodes report. Includes Policy Group, Policy and Policy Revision with desired format
Also adds some test cases in other areas where coverage is low
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: pkg/formatter/csv.go [hunk 1]
```diff
@@ -131,12 +131,17 @@ func MakeNodesReportCSV(records []*reporting.NodeReportItem, nodeFilter string)
 		)
 		if len(cookbooksList) != 0 {
 			cookbooksString = strings.Join(cookbooksList, " ")
+			if record.GetPolicyGroup() != "no group" {
```
Comment (tyler-ball) [inline pkg/formatter/csv.go:134]: I think having string comparison makes future refactors fragile. Instead, I think we should add a `HasPolicyGroup() Boolean` function

File: pkg/reporting/nodes.go [hunk 2]
```diff
@@ -83,6 +86,36 @@ func (nri *NodeReportItem) OSVersionPretty() string {
...
+}
+
+// GetPolicyWithRev returns formatted Policy name with revision
+func (nri *NodeReportItem) GetPolicyWithRev() string {
+
+	if nri.Policy == "" {
+		return "no policy"
+	}
+
+	return fmt.Sprintf("%s(rev %s)", nri.Policy, nri.PolicyRev)
```
Comment (tyler-ball) [inline pkg/reporting/nodes.go:116]:
```suggestion
	return fmt.Sprintf("%s (rev %s)", nri.Policy, nri.PolicyRev)
```
I think a space makes it more readable

File: pkg/formatter/csv.go (hunk 1 above)
Comment (mudash) [reply pkg/formatter/csv.go]: Agreed and fixed.

File: pkg/reporting/nodes.go (hunk 2 above)
Comment (mudash) [reply pkg/reporting/nodes.go]: done

**Comment by tyler-ball** [comment]
Looks great @mudash!

**Pull request merged**